### ISSUE TITLE
API fails to connect after some time.

### ISSUE THREAD

**Issue by shawntabrizi** [polkadot-js/api#1436]
When an application using the Polkadot-JS API is not connected for a long time, it fails to be able to connect at a later point until you restart the whole browser.

Hard to say exactly what is going on here, but I have seen this in multiple cases, across multiple people's computers.

Usually what is happening is that we are doing Substrate development. We connect a UI which uses the Polkadot-JS API (either Polkadot-JS Apps or `substrate-front-end-template`).

Upon initial connection to our local node, everything works as expected. Later the node is stopped so that users can do development.

Some time later, it may involve putting the computer to sleep (not sure), the user runs their local node again, and waits for the UI to connect to their node....

Nothing happens.

They refresh the page, nothing happens.

The error message is the standard one that it cannot find a node:

```
Provider.js:236 WebSocket connection to 'ws://127.0.0.1:9944/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
connect @ Provider.js:236
(anonymous) @ Provider.js:112
index.js:1375 2019-10-02 18:31:42          API-WS: disconnected from ws://127.0.0.1:9944 code: '1006' reason: ''
```

Only when you close the browser process, and then restart it, does it start working again, and connects no problem.

Both times I have seen it was using Chrome and on a Mac OSX computer.

Will be looking out for reproduction steps.

CC @hcastano

**Comment by shawntabrizi** [comment]
@jacogr Here is a video of it happening right now.

![image](https://user-images.githubusercontent.com/1860335/66171232-b8435e00-e647-11e9-9b16-fe7ab58a3da6.png)

There is an error message in the front-end-template when I did a restart and it DID connect to the node, but there was a bug in one of my components.

Either way, you see here that a node is running. The front-end-template does not see it.

I refresh... nothing.

I restart chrome and it connects no problem!

**Comment by jacogr** [comment]
The "Unknown Reason" is not helpful. However all browsers, due to security concerns, do not report any specific errors for WSS connection - as per RFC. So it can be, well, anything.

The fact that a page refresh doesn't connect but a re-open browser does, makes me think this is something OS-related.

**Comment by shawntabrizi** [comment]
The behaviors as I remember debugging before is that closing the _window_ and reopening it, not the process, will not fix the problem. Same with opening a tab. But I will get this all on video for you :)

**Comment by shawntabrizi** [comment]
@jacogr I just tried the new tab thing, and it DID recover... However, I am still pretty confident in the past it did not.

Either way, I feel pretty confident I can repo the issue for you if that helps. Just ping me :)

**Comment by jacogr** [comment]
So it is def some env. specific thing - so I ended up cloning the repo (after spending some time searching for it) and refreshing like crazy, and well, didn't get there. Sadly this is one of those things which, whatever it is seems there is nothing code-wise that can be done.

- we only use one WS connection
- we use the browser WS connection (no compat. layers like in Node)

So like this one https://github.com/polkadot-js/wasm/issues/17 - there is no known fix, maybe at some point it "magically disappears".

**Comment by jacogr** [comment]
Closing, not seen this in the wild at all.

**Comment by ical10** [comment]
Hello @jacogr , I am having the same problem but with a different error message: `Abnormal Closure`.

I am trying to connect to the API to get the balance of a user, but it always returned 0 (and from the node explorer it is confirmed that this user has a certain balance, not 0). Here is the video:
https://user-images.githubusercontent.com/14356826/117602386-b96d3300-b17a-11eb-82a0-c0a8f853aafa.mov

The error happened as I created an initial connection to the API. Nothing happened at this moment, balance could be fetched normally. A moment later (when my computer was put to sleep), I came back and this error started to appear. 

I tried to open a new tab, restart the browser, restart the laptop, nothing changed. 
Tried on Chrome/Firefox, nothing happened.

Here is the complete log:
```
index.js?ec14:378 WebSocket connection to 'ws://127.0.0.1:9944/' failed: 
connect @ index.js?ec14:378
connectWithRetry @ index.js?ec14:406
eval @ index.js?ec14:178
react_devtools_backend.js:2557 2021-05-10 10:11:18          
API-WS: disconnected from ws://127.0.0.1:9944: 1006:: Abnormal Closure
```

This error persists in my MacOS Mojave 10.14.6, using Firefox or Chrome/Brave browser. I am currently using v4.1.1 of Polkadot API.
